### PR TITLE
Release v0.3.0 - File-like streaming interface

### PR DESCRIPTION
**Pull request by kalouantonis**
Fixes issues #5, #9 and #10.

### CODE REVIEW

File: heatshrink/core.pyx [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+cdef class Writer:
+    """Thin wrapper around heatshrink_encoder"""
+    cdef _heatshrink.heatshrink_encoder *_hse
+
+    def __cinit__(self, **kwargs):
+        window_sz2 = kwargs.get('window_sz2', DEFAULT_WINDOW_SZ2)
+        lookahead_sz2 = kwargs.get('lookahead_sz2', DEFAULT_LOOKAHEAD_SZ2)
+
+        _validate_bounds(window_sz2, name='window_sz2',
+                        min=MIN_WINDOW_SZ2, max=MAX_WINDOW_SZ2)
```
Comment (matyasfodor) [inline heatshrink/core.pyx]: for me it seems, that this line should be one space more to the right.

File: heatshrink/core.pyx [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+        )
+        return res, sink_size
+
+    def poll(self, array.array out_buf):
+        """Poll data from state machine in to an array.
+
+        Assumes that the passed in array is large enough to
+        contain all data from the state machine.
+
+        FIXME: Protect against segfaults
```
Comment (matyasfodor) [inline heatshrink/core.pyx]: Isn't this fixed?

File: heatshrink/core.pyx (hunk 1 above)
Comment (kalouantonis) [reply heatshrink/core.pyx]: Yes, good catch

File: heatshrink/core.pyx (hunk 2 above)
Comment (kalouantonis) [reply heatshrink/core.pyx]: Yes, updated

**Comment by matyasfodor** [comment]
👍

**Review by matyasfodor — APPROVED** [review]

**Pull request merged**